### ISSUE TITLE
Maskgit transformer is hard to train

### ISSUE THREAD

**Issue by zhuqiangLu** [lucidrains/muse-maskgit-pytorch#18]
Hi
I am trying to train the transformer from scratch. I notice the loss quickly get stuck at a same value (my case is 8.x). I have tried lower/higher the learning rate but no luck. After ~15000k iters of training, the result is not satisfying as well.  
I have tried different masking ratio, such as ~34% instead of ~64%, the loss value does go down to ~1.3, but the outpointing result is still far from satisfying.

So, has anyone trained a working transformer?

**Comment by lucidrains** [comment]
@zhuqiangLu https://github.com/lucidrains/muse-maskgit-pytorch/commit/932087b5e7ee077ddd8cc65780d60b7e420f4953 join the discord and you can find others to collaborate with there, share your training experience etc

**Comment by lucidrains** [comment]
@zhuqiangLu also, get in touch with @patrickvonplaten , as huggingface expressed [interest](https://github.com/lucidrains/muse-maskgit-pytorch/issues/14) in training this

**Comment by ZeroCool940711** [comment]
I might be wrong but usually when the loss is too high the model is either not learning anything or overfitting, so, try decreasing the learning rate and see if that helps, a good value I found was 1e-7, you can start lower but after 1 epoch you should reduce it to have the model learn more fine details than the general shape of things.
A few things that I noticed from experience is that you should try to aim for a loss of 0.015-0.1 if possible, anything lower than that and you are overfitting, anything higher and the model can still learn more stuff, also, if you see the loss go up at a constant rate then you are already overfitting. The `vae loss` should always be lower than the `discr loss`, otherwise you are probably using a resolution that is too low compared to the `dim` and `vq_codebook_size` parameters, as I said, these things are from my experience, hope it's a bit useful.

**Comment by BinglengTang** [comment]
@ZeroCool940711 I have met the same problem. The loss in my experiment is going up at a constant learning rate at 1e-5. I am using a type pf vae that is used in dalle 1.0 and the text encoding in  my experiment is bert-large.

**Comment by lqniunjunlper** [comment]
same problem!!!
